src/event.cpp

```diff
@@ -178,26 +178,31 @@ void serialize_empty_rule(ddwaf_object &rule_map)
 }
 
 void serialize_and_consolidate_rule_actions(const ddwaf::rule &rule, ddwaf_object &rule_map,
-    action_type action_override, action_tracker &actions, ddwaf_object &stack_id)
+    std::string_view action_override, action_tracker &actions, ddwaf_object &stack_id)
 {
     const auto &rule_actions = rule.get_actions();
-    if (rule_actions.empty()) {
+    if (rule_actions.empty() && action_override.empty()) {
         return;
     }
 
     ddwaf_object tmp;
     ddwaf_object actions_array;
     ddwaf_object_array(&actions_array);
 
-    if (action_override == action_type::monitor) {
-        ddwaf_object_array_add(&actions_array, to_object(tmp, "monitor"));
+    if (!action_override.empty()) {
+        ddwaf_object_array_add(&actions_array, to_object(tmp, action_override));
+        auto action_it = actions.mapper.find(action_override);
+        if (action_it != actions.mapper.end()) {
+            const auto &[type, type_str, parameters] = action_it->second;
+            add_action_to_tracker(actions, action_override, type);
+        }
```

Comment:
* is it normal behavior that an unknown action is found? If not, this should log
* a little nit, but this block should do its actions in the same order as the code below  (`add_action_to_tracker`, then the unconditional `action_override`). Better if you could merge the two branches
* Is it normal this part doesn't have the `type == action_type::generate_stack` branch? If not, one more reason to merge the two branches

Reply:
> is it normal behavior that an unknown action is found? If not, this should log
This could be the case if an action ID doesn't have a definition, which can happen since these are provided through different configurations, so a race condition isn't unlikely.

> Is it normal this part doesn't have the type == action_type::generate_stack branch? If not, one more reason to merge the two branches
The expectation is that the exclusion filter override must be either `monitor`, `bypass` or a blocking action such as `block` or `redirect`, I can make this expectation more explicit during serialisation.

> a little nit, but this block should do its actions in the same order as the code below (add_action_to_tracker, then the unconditional action_override). Better if you could merge the two branches
Sure

Reply:
I changed the logic a bit to account for actions which perhaps weren't valid (non-blocking types).  https://github.com/DataDog/libddwaf/pull/303/commits/8c7666211cececa95436523849c34fc04d6cc2ea